### PR TITLE
Gene comparison tool beta

### PR DESCRIPTION
**Pull request by mattcgenui**

### CODE REVIEW

File: src/app/charts/box-plot/box-plot-view/box-plot-view.component.ts [hunk 1]
```diff
@@ -166,6 +166,9 @@ export class BoxPlotViewComponent implements OnInit, OnDestroy, AfterViewInit {
                 const currentGenes = self.dataService.getGeneEntries().slice().filter((g) => {
                     return g.model === self.geneService.getCurrentModel();
                 });
+
```
Comment (conundrumgirl) [inline src/app/charts/box-plot/box-plot-view/box-plot-view.component.ts:169]: a couple of console.logs left over from development?

File: src/app/genes/gene-comparison-tool/components/gene-comparison-tool-filter-list/gene-comparison-tool-filter-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+   styleUrls: [ './gene-comparison-tool-filter-list.component.scss' ],
+   encapsulation: ViewEncapsulation.None
+})
+export class GeneComparisonToolFilterListComponent {
+   @Input() filters: GeneComparisonToolFilter[];
+   @Output() onChange: EventEmitter<object> = new EventEmitter<object>();
+
+   hasSelectedFilters() {
+      for (const filter of this.filters) {
+         if (filter.options.filter(option => option.selected).length > 0) {
```
Comment (conundrumgirl) [inline src/app/genes/gene-comparison-tool/components/gene-comparison-tool-filter-list/gene-comparison-tool-filter-list.component.ts:17]: why not use 'find' here?

File: src/app/genes/gene-comparison-tool/components/gene-comparison-tool-filter-panel/gene-comparison-tool-filter-panel.component.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+export class GeneComparisonToolFilterPanelComponent {
+   @Input() filters: GeneComparisonToolFilter[];
+
+   isOpen: boolean = false;
+   activePane: number = -1;
+
+   @Output() onChange: EventEmitter<object> = new EventEmitter<object>();
+
+   toggleOption(option) {
+      option.selected = option.selected ? false : true;
```
Comment (conundrumgirl) [inline src/app/genes/gene-comparison-tool/components/gene-comparison-tool-filter-panel/gene-comparison-tool-filter-panel.component.ts]: options.selected= !option.selected seems simpler

**Review by conundrumgirl — APPROVED** [review]
A couple of small comments but didn't notice any issues otherwise

File: src/app/charts/box-plot/box-plot-view/box-plot-view.component.ts (hunk 1 above)
Comment (mattcgenui) [reply src/app/charts/box-plot/box-plot-view/box-plot-view.component.ts:169]: Thank you! I missed a couple...

File: src/app/genes/gene-comparison-tool/components/gene-comparison-tool-filter-list/gene-comparison-tool-filter-list.component.ts (hunk 2 above)
Comment (mattcgenui) [reply src/app/genes/gene-comparison-tool/components/gene-comparison-tool-filter-list/gene-comparison-tool-filter-list.component.ts:17]: Good suggestion!

File: src/app/genes/gene-comparison-tool/components/gene-comparison-tool-filter-panel/gene-comparison-tool-filter-panel.component.ts (hunk 3 above)
Comment (mattcgenui) [reply src/app/genes/gene-comparison-tool/components/gene-comparison-tool-filter-panel/gene-comparison-tool-filter-panel.component.ts]: You are right, I had forgotten !option.selected would return true if undefined!

**Pull request merged**